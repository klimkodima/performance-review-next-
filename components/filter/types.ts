export type FilterType = {
  name: string;
  value: boolean;
};
