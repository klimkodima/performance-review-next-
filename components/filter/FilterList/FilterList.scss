.filter-list {
  margin-top: 10px;

  &__name {
    display: flex;
    justify-content: space-between;
  }

  .plus-button {
    width: 20px;
    height: 20px;
    padding: 0;
    transform: scale(1.5);
    margin-right: 15px;
  }

  .title {
    color: #818181;
  }
}
