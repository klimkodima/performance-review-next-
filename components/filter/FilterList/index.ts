export { default as FilterList } from './FilterListContainer';
