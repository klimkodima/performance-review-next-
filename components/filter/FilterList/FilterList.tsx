import { ReactElement, MouseEvent, memo } from 'react';

import List from '@mui/material/List';
import Typography from '@mui/material/Typography';
import IconButton from '@mui/material/IconButton';
import { useTranslation } from 'next-i18next';

import FilterInput from '../FilterInput/FilterInput';
import { FilterOptionCheckbox } from '../FilterOptionCheckbox';
import { FilterPopUp } from '../FilterPopUp';
import { FilterListItem } from '../FilterListItem';

import { FilterType } from '../types';

import './FilterList.scss';

type FilterListPropsType = {
  popUpId: string | undefined;
  title: string;
  open: boolean;
  filter: FilterType[];
  anchorEl: HTMLButtonElement | null;
  onPopUpClose: () => void;
  onAddFilterClick: (event: MouseEvent<HTMLButtonElement>) => void;
};

const FilterList = memo(
  ({
    title,
    filter,
    onAddFilterClick,
    onPopUpClose,
    popUpId,
    open,
    anchorEl
  }: FilterListPropsType): ReactElement => {
    const { t } = useTranslation('header');
    return (
      <div className='filter-list'>
        <List data-testid='filter-list-test'>
          <div className='filter-list__name'>
            <Typography className='title'>{title}</Typography>
            <IconButton className='plus-button' onClick={onAddFilterClick}>
              +
            </IconButton>
          </div>
          {filter.map(
            ({ name, value }) =>
              value && <FilterListItem name={name} key={name} />
          )}
          <FilterPopUp
            anchorEl={anchorEl}
            popUpId={popUpId}
            handleClose={onPopUpClose}
            open={open}
          >
            {title === t('AUDITOR') && <FilterInput />}
            {filter.map(({ name, value }) => (
              <FilterOptionCheckbox name={name} value={value} key={name} />
            ))}
          </FilterPopUp>
        </List>
      </div>
    );
  }
);

export default FilterList;

FilterList.displayName = 'FilterList';
