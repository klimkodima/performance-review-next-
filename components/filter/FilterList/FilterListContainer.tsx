import { ReactElement, useState, MouseEvent } from 'react';
import FilterList from './FilterList';
import { FilterType } from '../types';

type FilterListContainerPropsType = {
  title: string;
  filter: FilterType[];
};

const FilterListContainer = ({
  filter,
  title
}: FilterListContainerPropsType): ReactElement => {
  const [anchorEl, setAnchorEl] = useState<HTMLButtonElement | null>(null);

  const handleAddFilterClick = (event: MouseEvent<HTMLButtonElement>) => {
    setAnchorEl(event.currentTarget);
  };

  const handlePopUpClose = () => {
    setAnchorEl(null);
  };

  const open = Boolean(anchorEl);
  const popUpId = open ? 'simple-popover' : undefined;

  return (
    <FilterList
      title={title}
      filter={filter}
      popUpId={popUpId}
      onAddFilterClick={handleAddFilterClick}
      onPopUpClose={handlePopUpClose}
      open={open}
      anchorEl={anchorEl}
    />
  );
};

export default FilterListContainer;
