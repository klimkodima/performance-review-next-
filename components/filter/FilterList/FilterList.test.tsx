import { render, screen } from '@testing-library/react';
import FilterList from './FilterList';
import { FilterType } from '../types';

const mockTeamFilter: FilterType[] = [
  { name: 'Alpha', value: true },
  { name: 'Charlie', value: false },
  { name: 'X-Rays', value: false }
];

describe('FilterList', () => {
  test('filter name shows', () => {
    render(
      <FilterList
        title='TEAM'
        open={true}
        popUpId='2'
        onPopUpClose={() => true}
        anchorEl={null}
        filter={mockTeamFilter}
        onAddFilterClick={() => true}
      />
    );
    expect(screen.getByText(/TEAM/i)).toBeInTheDocument();
  });

  test('Matches snapshot', () => {
    render(
      <FilterList
        title='TEAM'
        open={true}
        popUpId='2'
        onPopUpClose={() => true}
        anchorEl={null}
        filter={mockTeamFilter}
        onAddFilterClick={() => true}
      />
    );
    expect(screen.getByTestId('filter-list-test')).toMatchSnapshot();
  });
});
