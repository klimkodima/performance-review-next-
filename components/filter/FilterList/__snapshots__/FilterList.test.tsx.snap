// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterList Matches snapshot 1`] = `
<ul
  class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
  data-testid="filter-list-test"
>
  <div
    class="filter-list__name"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 title css-ahj2mt-MuiTypography-root"
    >
      TEAM
    </p>
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium plus-button css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      +
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
  <li
    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
  >
    <div
      class="filter-list-item"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 filter-list-item__filter-name css-ahj2mt-MuiTypography-root"
      >
        Alpha
      </p>
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium filter-list-item__delete-button css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        +
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </li>
</ul>
`;
